### PR TITLE
Switch Kokoro configuration to Bazel.

### LINKED ISSUE #642: Allow external developers to see Kokoro logs

### PR DESCRIPTION
**Pull request by coryan**
This fixes #642 (well this and a number of configuration changes).
We can grant access to the Kokoro logs to particular users (and 
eventually to all of the Internet).

This also free up a slot in the Travis build for more things we
cannot do in Kokoro, and also means we can upload the test results
to the ResultStore.

It also unblocks external projects (#518) for complicated reasons.

In summary, nothing major, but improves the CI a bit.

### CODE REVIEW

File: ci/kokoro/ubuntu/build.sh [hunk 1]
```diff
@@ -16,36 +16,100 @@
...
+    --keep_going \
+    --project_id=${KOKORO_PROJECT_ID} \
+    --auth_credentials="${KOKORO_GFILE_DIR}/build-results-service-account.json" \
+    --remote_instance_name="projects/${KOKORO_PROJECT_ID}" \
+    --invocation_id="${INVOCATION_ID}" \
+    --config=results-local \
+    -- //google/cloud/...:all
+
+echo
+echo "================================================================"
```
Comment (mbrukman) [inline ci/kokoro/ubuntu/build.sh]: Do we need double `=` lines, or can one suffice?

File: ci/kokoro/ubuntu/build.sh [hunk 2]
```diff
@@ -16,36 +16,100 @@
...
+build:results-local --auth_enabled=true
+build:results-local --auth_scope=https://www.googleapis.com/auth/cloud-source-tools
+build:results-local --experimental_remote_spawn_cache
+_EOF_
+
+# First build and run the unit tests.
+INVOCATION_ID="$(python -c 'import uuid; print uuid.uuid4()')"
+echo "Configure and start Bazel: " ${INVOCATION_ID}
+echo "================================================================"
+echo "    https://source.cloud.google.com/results/invocations/${INVOCATION_ID}"
```
Comment (mbrukman) [inline ci/kokoro/ubuntu/build.sh]: Any reason for the URL indentation, or can we make it flush-left?

File: ci/kokoro/ubuntu/build.sh [hunk 3]
```diff
@@ -16,36 +16,100 @@
...
+# echo "    Getting python modules"
+# sudo python2 -m pip install httpbin
+echo "End of download."
+
+echo ROOTS PATH = ${GRPC_DEFAULT_SSL_ROOTS_FILE_PATH}
 
-# The integration tests
-export GRPC_DEFAULT_SSL_ROOTS_FILE_PATH="${PROJECT_ROOT}/third_party/grpc/etc/roots.pem"
+ls -l ${GRPC_DEFAULT_SSL_ROOTS_FILE_PATH} || echo "no roots file"
+ls -l $(dirname ${GRPC_DEFAULT_SSL_ROOTS_FILE_PATH}) || echo "no roots file dir"
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: Fixed.

File: ci/kokoro/ubuntu/build.sh [hunk 4]
```diff
@@ -16,36 +16,100 @@
...
+go get -u cloud.google.com/go/bigtable/cmd/cbt
+# echo "    Getting python modules"
+# sudo python2 -m pip install httpbin
+echo "End of download."
+
+echo ROOTS PATH = ${GRPC_DEFAULT_SSL_ROOTS_FILE_PATH}
 
-# The integration tests
-export GRPC_DEFAULT_SSL_ROOTS_FILE_PATH="${PROJECT_ROOT}/third_party/grpc/etc/roots.pem"
+ls -l ${GRPC_DEFAULT_SSL_ROOTS_FILE_PATH} || echo "no roots file"
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: Fixed.

File: ci/kokoro/ubuntu/build.sh [hunk 5]
```diff
@@ -16,36 +16,100 @@
...
+cp "$(bazel info output_base)/java.log" "${KOKORO_ARTIFACTS_DIR}/" || echo "java log copy failed."
+echo "End of copying."
+
+echo
+echo "================================================================"
+echo "================================================================"
+# Then build everything else (integration tests, examples, etc). This needs to
+# go last (and in a separate invocation) so Bazel state is preserved to run the
+# integration tests.
+bazel build \
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: Fixed.  Users running this script manually would be hard, they can just call bazel directly, but this script assumes it is running in the CI environment.

File: ci/kokoro/ubuntu/build.sh [hunk 6]
```diff
@@ -16,36 +16,100 @@
...
 
+echo
+echo "================================================================"
+echo "================================================================"
 echo "Running Google Cloud Bigtable Integration Tests"
-(cd google/cloud/bigtable/tests && \
-    "${PROJECT_ROOT}/google/cloud/bigtable/tests/run_integration_tests_production.sh")
+(cd $(bazel info bazel-bin)/google/cloud/bigtable/tests && \
+   "${PROJECT_ROOT}/google/cloud/bigtable/tests/run_integration_tests_production.sh")
+(cd $(bazel info bazel-bin)/google/cloud/bigtable/tests && \
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: Unfortunately this is correct, added TODO and bug (#730) to explain.

File: ci/kokoro/ubuntu/build.sh [hunk 7]
```diff
@@ -16,36 +16,100 @@
...
+echo "================================================================"
+export GRPC_DEFAULT_SSL_ROOTS_FILE_PATH="$(bazel info output_base)/external/com_github_grpc_grpc/etc/roots.pem"
+export GOOGLE_APPLICATION_CREDENTIALS="${KOKORO_GFILE_DIR}/service-account.json"
+
+echo "Download dependencies for integration tests."
+echo "    Getting cbt tool"
+export GOPATH="${KOKORO_ROOT}/golang"
+go get -u cloud.google.com/go/bigtable/cmd/cbt
+# echo "    Getting python modules"
+# sudo python2 -m pip install httpbin
```
[the remark this answers is not in the document]
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: Fixed.

File: ci/kokoro/ubuntu/build.sh (hunk 1 above)
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: The log files, specially in Sponge, are much easier to follow with the double `=` lines.

File: ci/kokoro/ubuntu/build.sh (hunk 2 above)
Comment (coryan) [reply ci/kokoro/ubuntu/build.sh]: Fixed (also it was easier to find in the log).

File: ci/kokoro/ubuntu/build.sh [hunk 8]
```diff
@@ -16,36 +16,110 @@
...
+build:results-local --bes_best_effort=false
+build:results-local --bes_timeout=10m
+build:results-local --tls_enabled=true
+build:results-local --auth_enabled=true
+build:results-local --auth_scope=https://www.googleapis.com/auth/cloud-source-tools
+build:results-local --experimental_remote_spawn_cache
+_EOF_
+
+# First build and run the unit tests.
+INVOCATION_ID="$(python -c 'import uuid; print uuid.uuid4()')"
```
Comment (mbrukman) [inline ci/kokoro/ubuntu/build.sh]: `readonly`?

File: ci/kokoro/ubuntu/build.sh [hunk 9]
```diff
@@ -16,36 +16,110 @@
...
+    --keep_going \
+    -- //google/cloud/...:all
+
+# The integration tests need further configuration and tools.
+echo
+echo "================================================================"
+echo "================================================================"
+
+echo "Download dependencies for integration tests."
+# Download the gRPC roots.pem file, somewhere inside the bowels of Bazel this
```
Comment (mbrukman) [inline ci/kokoro/ubuntu/build.sh]: Propose rewriting as:

```
# Download the gRPC `roots.pem` file. Somewhere inside the bowels of Bazel, this
# file might exist, but my attempts at using it have failed.
```

**Pull request merged**